src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.cs

```diff
@@ -59,6 +64,12 @@ private static async Task InitializeStandardComponentServicesAsync(HttpContext h
             authenticationStateProvider.SetAuthenticationState(Task.FromResult(authenticationState));
         }
 
+        var formData = httpContext.RequestServices.GetRequiredService<FormDataProvider>() as IHostEnvironmentFormDataProvider;
+        if (handler != null && formData != null)
+        {
+            formData.SetFormData(handler, form!.ToDictionary(kvp => kvp.Key, kvp => kvp.Value[0]).AsReadOnly());
```

Comment:
How do we know `form` isn't `null` here?

Reply:
Nope. This was VS "helping"